### PR TITLE
feat: add Pausable.sol

### PR DESCRIPTION
**Pull request by calnix**

### CODE REVIEW

File: foundry/src/Pausable.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * @dev Contract module which allows children to implement an emergency stop
+ * mechanism that can be triggered by an authorized account.
+ *
+ * This module is used through inheritance. It will make available the
+ * modifiers `whenNotPaused` and `whenPaused`, which can be applied to
+ * the functions of your contract. Note that they will not be pausable by
+ * simply including this module, only once the modifiers are put in place.
+ */
+
+contract Pausable is AccessControl {
```
Comment (alcueca) [inline foundry/src/Pausable.sol:16]: It will be funny to see how the inheritance tree works if a Wand inherits from both Pausable and AccessControl

**Review by alcueca — CHANGES_REQUESTED** [review]
The fixes required are pretty minor.

File: foundry/src/Pausable.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  /// @dev Returns to active state. Requires: contract must be paused
+  function unpause() external auth whenPaused {
+    paused = false;
+    emit PausedState(msg.sender, paused);
+  }
+
+  /// @dev Modifier to make a function callable only when the contract is not paused
+  modifier whenNotPaused() {
+    require(paused == false, "Pausable: paused");
```
[the remark this answers is not in the document]
Comment (calnix) [reply foundry/src/Pausable.sol:42]: Just wanted to confirm, do we really want to switch to using custom errors - when we adhere to require statements everywhere else?

[the remark this answers is not in the document]
Comment (alcueca) [reply foundry/src/Pausable.sol:42]: Yeah, we are switching to errors every time we refactor something.

File: foundry/src/Pausable.sol [hunk 3]
```diff
@@ -14,7 +14,10 @@ import "@yield-protocol/utils-v2/access/AccessControl.sol";
  */
 
 contract Pausable is AccessControl {
-    
+
+  error requirePaused(address msgSender, bool pausedState);
```
Comment (alcueca) [inline foundry/src/Pausable.sol:18]: Errors are UpperCamelCase

**Review by alcueca — APPROVED** [review]
Minor thing, please correct the naming and then merge.

**Pull request merged**